coalib/results/TextRange.py

```diff
@@ -89,7 +89,19 @@ def end(self):
         return self._end
 
     def overlaps(self, other):
-        return self.start <= other.end and self.end >= other.start
+        return self.is_behind(self.start, other.end) and \
+            self.is_behind(other.start, self.end)
+
+    @staticmethod
+    def is_behind(first_position, second_position):
+        if first_position.line is None or second_position.line is None:
+            return True
+        if first_position.line == second_position.line:
+            return True if None in \
```

Comment:
Stylistic thing, but I think we prefer parenthesis syntax over the escaped char one `\` (like return(...))